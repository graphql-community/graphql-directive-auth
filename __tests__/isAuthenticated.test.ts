import { graphql } from 'graphql';
import MockExpressRequest from 'mock-express-request';
import AuthDirective from '../src/index';
import schema from '../example/schema';

beforeAll(() => {
  process.env.APP_SECRET = '123';
});

test('getDirectiveDeclaration should be defined', () => {
  expect(
    AuthDirective().isAuthenticated.getDirectiveDeclaration()
  ).toMatchSnapshot();
});

test('if call resolver if Authorization header is set to correct value', () =>
  graphql(
    schema,
    `
      query {
        me {
          id
          username
          isAdmin
        }
      }
    `,
    {},
    {
      req: new MockExpressRequest({
        headers: {
          Authorization:
            'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbG8iOjEyMywiaWF0IjoxNTMyNzI3MDMzfQ.nSGh5q9YXZd-pWYagOxD2f-9hdNPw08e4MnKASgn4wY',
        },
      }),
    }
  ).then(response => {
    expect(response).toMatchSnapshot({
      data: {
        me: {
          id: expect.any(String),
          username: expect.any(String),
          isAdmin: expect.any(Boolean),
        },
      },
    });
  }));

test('if throw an Error if Authorization header is not set to correct value', () =>
  graphql(
    schema,
    `
      query {
        me {
          id
          username
          isAdmin
        }
      }
    `,
    {},
    {
      req: new MockExpressRequest({ headers: {} }),
    }
  ).then(response => {
    expect(response.errors[0].message).toEqual('Not authorized!');
    expect(response.data.me).toBeNull();
  }));
