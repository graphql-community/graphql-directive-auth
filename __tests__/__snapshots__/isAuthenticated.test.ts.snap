// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDirectiveDeclaration should be defined 1`] = `
Object {
  "args": Object {},
  "astNode": undefined,
  "description": undefined,
  "extensions": undefined,
  "isRepeatable": false,
  "locations": Array [
    "FIELD_DEFINITION",
  ],
  "name": "isAuthenticated",
}
`;

exports[`if call resolver if Authorization header is set to correct value 1`] = `
Object {
  "data": Object {
    "me": Object {
      "id": Any<String>,
      "isAdmin": Any<Boolean>,
      "username": Any<String>,
    },
  },
}
`;
